pkg/login/auth.go

```diff
@@ -25,14 +27,34 @@ var (
 
 var loginLogger = log.New("login")
 
-var AuthenticateUserFunc = AuthenticateUser
+type Authenticator interface {
+	AuthenticateUser(context.Context, *models.LoginUserQuery) error
+}
+
+var AuthenticateUserFunc = func(context.Context, *models.LoginUserQuery) error {
```

Comment:
Do we need this? I'm checking the code and I found an occurrence inside `login.go` file and we can change directly with the authenticator call.